Networkscan Repo Cleanup

apurvagoenka-method opened the pull request:
# Reorganize CLI commands into discover, enumerate, and pentest modules

This PR restructures the CLI command organization into three main modules:

1. **discover** - Commands for discovering hosts, ports, services, and TLS information
2. **enumerate** - Commands for enumerating information about network services (FTP, SMTP, SSH)
3. **pentest** - Commands for penetration testing (currently bruteforce attacks)

The reorganization improves command structure and naming consistency while maintaining all existing functionality. The Fern schema definitions were also reorganized to match the new command structure.

apurvagoenka-method commented:
* **#62** <a href="https://app.graphite.dev/github/pr/method-security/networkscan/62?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#61** <a href="https://app.graphite.dev/github/pr/method-security/networkscan/61?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#60** <a href="https://app.graphite.dev/github/pr/method-security/networkscan/60?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/method-security/networkscan/60?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

payoub125 commented on cmd/discover.go:
```diff
@@ -0,0 +1,210 @@
+		},
+	}
+	discoverHostCmd.Flags().String("target", "", "Target IP, host, or CIDR to scan for hosts")
+	discoverHostCmd.Flags().String("scan-type", "", "Scan type for host discovery (tcpsyn | tcpack | icmpecho | icmptimestamp | arp | icmpaddressmask)")
```
Do we want to ENUM these?

apurvagoenka-method replied:
added

payoub125 commented on cmd/discover.go:
```diff
@@ -0,0 +1,210 @@
+				a.OutputSignal.AddError(err)
+				return
+			}
+			report, err := discover.RunServiceFingerprint(cmd.Context(), timeout, target, port)
```
lets make this a config with a fern def

apurvagoenka-method replied:
improved!

payoub125 commented on cmd/discover.go:
```diff
@@ -0,0 +1,210 @@
+				a.OutputSignal.AddError(err)
+				return
+			}
+			report, err := discover.GetTLSInfo(cmd.Context(), targets, discoverFern.DiscoverTlsConfig{Timeout: timeout, InsecureSkipVerify: insecure})
```
lets use the helper function on line 203

apurvagoenka-method replied:
great catch, improved!

payoub125 commented on cmd/discover.go:
```diff
@@ -0,0 +1,210 @@
+		Short: "Fingerprint the operating system on a target host",
+		Long:  `Fingerprint the operating system on a target host`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if !privileges.IsPrivileged {
```
I wonder if this still works?

apurvagoenka-method replied:
it does!
```
> go run main.go discover host --target 192.168.1.1
{"content":"bnVsbA==","started_at":"2025-05-19T15:54:01.547469-04:00","completed_at":"2025-05-19T15:54:01.548014-04:00","status":1,"error_message":"discover host can only be run as a privileged user"}
```

payoub125 commented:
Looks good! Left a couple comments. Also just double checking we tested all these commands after the update

apurvagoenka-method commented:
@payoub125 addressed comments and yes double checked everything since new signals were gen'ed for the Ontology [PR](https://github.com/method-security/ontology/pull/93)

payoub125 commented:
> @payoub125 addressed comments and yes double checked everything since new signals were gen'ed for the Ontology [PR](https://github.com/method-security/ontology/pull/93)

Sweet looks awesome!

apurvagoenka-method commented on .github/workflows/reusable-build.yml:
```diff
@@ -16,61 +16,86 @@
+          path: dist/
+          retention-days: 7
+
+  prepare-linux-amd64:
```
Resolve
